tools/test/donkeyrider.py

```diff
@@ -0,0 +1,133 @@
+                        help='build images, instead of only downloading')
+    parser.add_argument('-f', '--filter', dest='filter_tests', type=str, action='append',
+                        help='filter tests')
+    parser.add_argument('-d', '--dist', dest='dist', type=str, help='select distribution')
+    parser.add_argument('--py', '--python', dest='python', type=str, help='select Python version')
+    parser.add_argument('-s', '--suite', dest='suite', type=str, help='select test suite')
+    parser.add_argument('--db', '--database', dest='database', type=str, help='select RDBMS')
+    args = parser.parse_args()
+    if args.test_matrix_file is None:
+        matrix_file = os.path.join(project_dir, "etc/docker/test/matrix.yml")
+    else:
+        matrix_file = args.test_matrix_file
+    print("Using test matrix from file", matrix_file, file=sys.stderr)
+
+    # make sure the following imports work
+    sys.path.append(os.path.abspath(os.path.dirname(__file__)))
+
+    # matrix parsing
+
+    from matrix_parser import parse_matrix
+
+    with open(matrix_file, 'r') as fhandle:
+        test_matrix = parse_matrix(fhandle)
+
+    if not test_matrix:
+        print("Matrix could not be determined", file=sys.stderr)
+        sys.exit(1)
+
+    def argument_filter(case):
+        if args.dist:
+            if case['DIST'] != args.dist:
+                return False
+        if args.python:
+            if case['PYTHON'] != args.python:
+                return False
+        if args.suite:
+            if case['SUITE'] != args.suite:
+                return False
+        if args.database:
+            if case['RDBMS'] != args.database:
+                return False
+        return True
+
+    test_matrix = list(filter(argument_filter, test_matrix))
+
+    if len(test_matrix) == 0:
+        print("No more test cases after filtering. Exiting..", file=sys.stderr)
+        sys.exit(0)
+    if len(test_matrix) == 1 and 'PARALLEL_AUTOTESTS' not in os.environ:
+        print("One test case remaining after filtering. Switching to serial run", file=sys.stderr)
+        os.environ['PARALLEL_AUTOTESTS'] = "0"
+    elif 'PARALLEL_AUTOTESTS' not in os.environ or (os.environ['PARALLEL_AUTOTESTS'] not in ['false', '0']):
+        print("Tests will run in parallel", file=sys.stderr)
+        os.environ['PARALLEL_AUTOTESTS'] = "1"
+
+    # image building
+
+    from build_images import build_main
+
+    build_args = [os.path.join(project_dir, "etc/docker/test")]
+    if not args.build:
+        build_args.insert(0, "--download-only")
+    images = build_main(test_matrix, build_args)
+
+    if not images:
+        print("Images could not be built", file=sys.stderr)
+
+    # test running
+
+    from run_tests import run_tests
+
+    run_tests(test_matrix, images, args.filter_tests)
+
+
+if __name__ == "__main__":
+    signal.signal(signal.SIGINT, shutdown)
+    signal.signal(signal.SIGTERM, shutdown)
+    main()
```

Comment:
The name `donkeyrider` is very creative, and I like it, however it is even more un-descriptive. I would not expect someone new to the project asking themself: what is this and can I eat it?

How about `run_test_pipeline` or `run_test_suite` or ...

Reply:
As a developer tool I think it is fine. I was thinking the same that it would be obscure, but somehow we sticked to it. I don't think it is necessary to change the name